### PR TITLE
add network integration test

### PR DESCRIPTION
**Pull request by futuretea**
#388

### CODE REVIEW

File: tests/integration/api/network_apis_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		scaled        *config.Scaled
+		nadController ctlcniv1.NetworkAttachmentDefinitionController
+		nadNamespace  string
+		networkName   string
+	)
+
+	BeforeEach(func() {
+		scaled = harvester.Scaled()
+		nadController = scaled.CniFactory.K8s().V1().NetworkAttachmentDefinition()
+		nadNamespace = "default"
```
Comment (gitlawr) [inline tests/integration/api/network_apis_test.go]: Seems like we don't need to initialize for each test

File: tests/integration/api/network_apis_test.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	})
+
+	Cleanup(func() {
+		nadList, err := nadController.List(nadNamespace, metav1.ListOptions{
+			LabelSelector: labels.FormatLabels(testResourceLabels)})
+		if err != nil {
+			GinkgoT().Logf("failed to list tested networks, %v", err)
```
Comment (gitlawr) [inline tests/integration/api/network_apis_test.go]: s/tested/test/

File: tests/integration/api/network_apis_test.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+	var (
+		scaled        *config.Scaled
+		nadController ctlcniv1.NetworkAttachmentDefinitionController
+		networkName   string
+	)
+
+	BeforeEach(func() {
+		scaled = harvester.Scaled()
+		nadController = scaled.CniFactory.K8s().V1().NetworkAttachmentDefinition()
```
Comment (gitlawr) [inline tests/integration/api/network_apis_test.go:71]: I mean we can move these out of the BeforeEach function

File: tests/integration/api/network_apis_test.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		}
+	})
+
+	Context("operate via steve API", func() {
+
+		var networkAPI string
+
+		BeforeEach(func() {
+
+			networkAPI = helper.BuildAPIURL("v1", "k8s.cni.cncf.io.network-attachment-definitions")
```
Comment (gitlawr) [inline tests/integration/api/network_apis_test.go:94]: and this?

File: tests/integration/api/network_apis_test.go (hunk 3 above)
Comment (futuretea) [reply tests/integration/api/network_apis_test.go:71]: If move variables  these out, the values of these variables will be assigned before all tests are executed.

File: tests/integration/api/network_apis_test.go (hunk 4 above)
Comment (futuretea) [reply tests/integration/api/network_apis_test.go:94]: If move it out, networkAPI will become `https://localhost:0/v1/k8s.cni.cncf.io.network-attachment-definitions`

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**